pkg/gke/components/CruGKE.vue

```diff
@@ -0,0 +1,840 @@
+      this.$set(this.normanCluster.gkeConfig, 'privateClusterConfig', cloneDeep(defaultGkeConfig.privateClusterConfig));
+    }
+    this.config = this.normanCluster.gkeConfig;
+    this.nodePools = this.normanCluster.gkeConfig.nodePools;
+
+    this.nodePools.forEach((pool) => {
+      this.$set(pool, '_id', randomStr());
+      this.$set(pool, '_isNewOrUnprovisioned', this.isNewOrUnprovisioned);
+      if (!pool.management) {
+        this.$set(pool, 'management', {});
+      }
+      if (!pool.autoscaling) {
+        this.$set(pool, 'autoscaling', {});
+      }
+    });
+  },
+
+  data() {
+    const store = this.$store as Store<any>;
+    const supportedVersionRange = store.getters['management/byId'](MANAGEMENT.SETTING, SETTING.UI_SUPPORTED_K8S_VERSIONS)?.value;
+
+    return {
+      normanCluster:    { name: '' } as any,
+      nodePools:        [] as GKENodePool[],
+      config:           { } as GKEConfig,
+      membershipUpdate: {} as any,
+      originalVersion:  '',
+      defaultImageType: imageTypes[0],
+      supportedVersionRange,
+
+      loadingMachineTypes:  false,
+      machineTypesResponse: {} as getGKEMachineTypesResponse,
+
+      fvFormRuleSets: [
+        {
+          path:  'diskSizeGb',
+          rules: ['diskSizeGb']
+        },
+        {
+          path:  'initialNodeCount',
+          rules: ['initialNodeCount']
+        },
+        {
+          path:  'ssdCount',
+          rules: ['ssdCount']
+        },
+        {
+          path:  'nodeGeneral',
+          rules: ['minMaxNodeCount', 'nodePoolNameUnique']
+        },
+        {
+          path:  'poolName',
+          rules: ['poolNameRequired']
+        },
+        {
+          path:  'masterIpv4CidrBlock',
+          rules: ['masterIpv4CidrBlockRequired', 'masterIpv4CidrBlockFormat']
+        },
+        {
+          path:  'clusterName',
+          rules: ['nameRequired', 'clusterNameChars', 'clusterNameStartEnd']
+        },
+        {
+          path:  'clusterIpv4CidrBlock',
+          rules: ['clusterIpv4CidrBlockFormat']
+        },
+        {
+          path:  'nodeIpv4CidrBlock',
+          rules: ['nodeIpv4CidrBlockFormat']
+        },
+        {
+          path:  'servicesIpv4CidrBlock',
+          rules: ['servicesIpv4CidrBlockFormat']
+        },
+        {
+          path:  'clusterIpv4Cidr',
+          rules: ['clusterIpv4CidrFormat']
+        },
+      ],
+      isAuthenticated: false,
+
+      debouncedLoadGCPData: () => {}
+
+    };
+  },
+
+  created() {
+    if (!this.nodePools.length) {
+      this.addPool();
+    }
+    const registerBeforeHook = this.registerBeforeHook as Function;
+    const registerAfterHook = this.registerAfterHook as Function;
+
+    registerBeforeHook(this.cleanPoolsForSave);
+    registerBeforeHook(this.removeUnchangedConfigFields);
+    registerAfterHook(this.saveRoleBindings, 'save-role-bindings');
+
+    this.debouncedLoadGCPData = debounce(this.loadGCPData, 500);
```

Comment:
I think happened here was that I put it into `created()` because that seems to be how we use debounce across the app, and I wanted to stick to the pattern. Then typescript complained at me, so I added a definition to data as well...moving to data would _probably_ be fine.  But I like the consistency with other uses of debounce around the dashboard.